docs/iam-usage.rst

```diff
@@ -0,0 +1,162 @@
+-  A Google Service Account: ``iam.service_account(email)``, or ``'serviceAccount:{email}'``.
+-  A Google Group: ``iam.group(email)``, or ``'group:{email}``.
+-  A Google Apps domain: ``iam.domain(domain_name)``, or ``'domain:{email}``.
+-  Any authenticated Google user: ``iam.ALL_AUTHENTICATED_USERS``, or ``'allAuthenticatedUsers'``.
+-  Anyone: ``iam.ALL_USERS``, or ``'allUsers'``.
+
+.. note::
+  All of these are convenience wrappers around strings.
+  See the list of member string formats `here <https://cloud.google.com/iam/docs/managing-policies>`_.
+
+Roles
+~~~~~
+
+Roles represent bundles of permissions that can be added to members.
+For a complete list of roles available on a resource run::
+
+    >>> resource.get_roles()
+
+An ``iam.Role`` object has a name, title, and description
+
+- ``name``: the canonical name of a role. This will be the value
+  used as keys in policy dictionaries (see below), and will be
+  referred to as the "role string" throughout this document.
+  E.g. ``'roles/owner'``.
+- ``title``: human readable title of the role. E.g. ``'Owner'``
+- ``description``: the description of a role.
+
+Policies
+~~~~~~~~
+
+A policy consists of three items, which are returned as a tuple by ``resource.get_policy()`` (see below):
+
+- A policy dictionary, where the keys are role strings, and the values are ``set`` s of member strings.
+- A version. An integer for end-user bookkeeping.
+- An etag used to provide optimistic concurrency controls on policy updates.
+
+
+Methods
+----------------------------------
+
+Resources that implement the IAM interface provide the following methods:
+
+``get_policy`` returns a tuple of ``(policy, version, etag)`` on the corresponding resource.
+
+>>> policy, version, etag = resource.get_policy()
+>>> policy
+{
+   'roles/owner': set(['user:alice@example.com']),
+   'roles/editor: set(['group:admins@example.com']),
+   'roles/reader': set(['domain:example.com', 'user:bob@example.com'])
+
+- A policy dictionary, where the keys are role strings, and the values are ``set`` s of member strings.
+- A version. An integer for end-user bookkeeping.
+- An etag used to provide optimistic concurrency controls on policy updates.
+
+
+Methods
+----------------------------------
+
+Resources that implement the IAM interface provide the following methods:
+
+``get_policy`` returns a tuple of ``(policy, version, etag)`` on the corresponding resource.
+
+>>> policy, version, etag = resource.get_policy()
+>>> policy
+{
+   'roles/owner': set(['user:alice@example.com']),
+   'roles/editor: set(['group:admins@example.com']),
+   'roles/reader': set(['domain:example.com', 'user:bob@example.com'])
+}
+>>> version
+0
+>>> etag
+ffdFADFdsgfsjrsHTY
+
+``set_policy`` takes one positional, and three keyword arguments:
+
+- The first argument is a policy dictionary, described above.
+- ``version`` is an optional integer. If ommited the version of the policy will be set to 0
```

Comment:
I also think in general, going through each parameter  belong more in the doc strings, usage docs, I would mostly just demonstrate by example here.